user/waydroid/patches/sed-chimera.patch

```diff
@@ -0,0 +1,17 @@
+--- a/tools/helpers/lxc.py	2024-09-07 13:39:38.419147354 -0300
++++ b/tools/helpers/lxc.py	2024-09-07 14:10:55.719543931 -0300
+@@ -157,12 +157,12 @@
+     tools.helpers.run.user(args, command)
+     command = ["sh", "-c", "cat {} > \"{}\"".format(' '.join('"{0}"'.format(w) for w in config_snippets), lxc_path + "/config")]
+     tools.helpers.run.user(args, command)
+-    command = ["sed", "-i", "s/LXCARCH/{}/".format(platform.machine()), lxc_path + "/config"]
++    command = ["sed", "-i", "\'\'","s/LXCARCH/{}/".format(platform.machine()), lxc_path + "/config"]
```

Comment:
@tulilirockz is this actually correct? seems to me like it should be more like `"sed", "-i", "", "..."` (otherwise you will get backup files that are called `''`) unless there is something i'm missing?

Reply:
(i'd test this myself but i have no idea how this is supposed to work, so just something i spotted while looking over the diff - so if this is intentional, nevermind me)

Reply:
Yup! It works fine. Apparent bsdsed needs to have an argument specifiying the empty extension. If you just gave it an empty arg it would thing the `s/*` command is the backup extension, like:

```
sed -i s/thing/thong/g thing.txt
sed -i "" -> sed -i (empty space) s/thing/thong #...
# where as '' gets expanded to actual single quotes
sed -i '' -> sed -i (no extension) s/thing/thong
```

Reply:
i think you misunderstand:

```
q66@chimera: /home/q66$ cat test.txt
foo
q66@chimera: /home/q66$ sed -i '' 's/foo/bar/' test.txt
q66@chimera: /home/q66$ cat test.txt                   
bar
q66@chimera: /home/q66$ sed -i "" 's/bar/baz/' test.txt
q66@chimera: /home/q66$ cat test.txt                   
baz
q66@chimera: /home/q66$ sed -i "''" 's/baz/foo/' test.txt
q66@chimera: /home/q66$ cat test.txt*
foo
baz
q66@chimera: /home/q66$ cat test.txt
test.txt      test.txt\'\'
```

yes, you need to pass an empty argument to `-i` to get no backup, but you are already doing so if you pass `""`

if you pass `"\'\'"`, you are passing two single quotes as a backup extension, which will result in files with extensions being run

you can see this if you do it from python:

```
>>> import subprocess
>>> subprocess.run(["sed", "-i", "", "s/foo/bar/", "test.txt"])
CompletedProcess(args=['sed', '-i', '', 's/foo/bar/', 'test.txt'], returncode=0)
>>> subprocess.run(["sh", "-c", "ls -l test.txt*"])
-rw-r--r--  1 q66 q66 4 Sep  8 21:39 test.txt
CompletedProcess(args=['sh', '-c', 'ls -l test.txt*'], returncode=0)
>>> subprocess.run(["sed", "-i", "\'\'", "s/foo/bar/", "test.txt"])
CompletedProcess(args=['sed', '-i', "''", 's/foo/bar/', 'test.txt'], returncode=0)
>>> subprocess.run(["sh", "-c", "ls -l test.txt*"])
-rw-r--r--  1 q66 q66 4 Sep  8 21:40 test.txt
-rw-r--r--  1 q66 q66 4 Sep  8 21:39 test.txt''
CompletedProcess(args=['sh', '-c', 'ls -l test.txt*'], returncode=0)
```

that is unless the `tools.helpers.run.user` has different semantics here, which is what i am asking (as it does not seem to be the case)

Reply:
as far as i can tell, the command is just passed to `subprocess.Popen` which has the same semantics as `subprocess.run`, so i should be correct here

Reply:
oh my god you are right, it literally just generated the `''` files as you said :O, i didnt see them earlier

Reply:
```
~ 
❯ ls /var/lib/waydroid/lxc/waydroid/
╭───┬─────────────────────────────────────────────────┬──────┬─────────┬─────────────╮
│ # │                      name                       │ type │  size   │  modified   │
├───┼─────────────────────────────────────────────────┼──────┼─────────┼─────────────┤
│ 0 │ /var/lib/waydroid/lxc/waydroid/config''         │ file │   955 B │ 2 hours ago │
│ 1 │ /var/lib/waydroid/lxc/waydroid/config           │ file │   954 B │ 2 hours ago │
│ 2 │ /var/lib/waydroid/lxc/waydroid/waydroid.seccomp │ file │   346 B │ 3 hours ago │
│ 3 │ /var/lib/waydroid/lxc/waydroid/config_nodes     │ file │ 1.5 KiB │ 2 hours ago │
│ 4 │ /var/lib/waydroid/lxc/waydroid/config_session   │ file │   313 B │ an hour ago │
╰───┴─────────────────────────────────────────────────┴──────┴─────────┴─────────────╯
```

Reply:
ok :) you can make another PR to fix that then

Reply:
alright!